src/common/multiline-ellipsis/index.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Adds an ellipsis to a el based on the height and width
+ * specified in CSS rules
+ */
+function truncate(el) {
+    if (!el) {
+        return;
+    }
+
+    const text = el.innerText;
+
+    if (this.isTextOverflowing(el)) {
+        const checkFunc = i => {
+            el.innerText = text.substring(0, i);
```

Comment:
I think it should be safe to switch to `textContent` instead of `innerText` which is going to be faster since it `innerText` requires layout to be completed.

Reply:
Let me try it out (for the sake of speed).

My only worry is whether someone adds any hidden text to a button (not saying it's going to happen, but someone might throw something in there):
>`innerText` is aware of styling and won’t return the text of “hidden” elements, whereas `textContent` does.

Reply:
So far, in my tests, I'm not seeing a material performance gain using `textContent`.

Also, from my tests, my worry about pulling in hidden content is actually unfounded. Therefore, it comes down to the performance difference between `innerText` and `textContent`, of which I can't confirm the improvement.

Reply:
It can be a bit of a pain to profile this but `textContent` is definitely much faster in real world apps. When you have `innerText` reflows *must* happen before you can read the value which can cause jank. `textContent` on the other hand does not have this issue.

Relevant excerpt from the MDN link:
>Since innerText takes into account CSS styling, reading this property triggers a reflow, ensuring up-to-date computed style. textContent doesn’t. Reflows can be computationally expensive thus should be avoided when possible.

Reply:
Updated per your feedback, @DylanPiercey.